NumberBaseball/NumberBaseball/main.swift

```diff
@@ -1,67 +1,101 @@
-        ballCount += computerNumberList.contains(randomNumberList[index]) ? 1 : 0
+        strike += computerNumberList[index] == userNumberList[index] ? 1 : 0
+        ball += computerNumberList.contains(userNumberList[index]) ? 1 : 0
     }
     
-    ballCount -= strikeCount
-    return (strikeCount, ballCount)
+    ball -= strike
+    print("""
+          임의의 수 : \(userNumberList[0]) \(userNumberList[1]) \(userNumberList[2])
+          \(strike) 스트라이크, \(ball) 볼
+          """)
+    return (strike, ball)
 }
 
-func isGameOver(_ strikeCount: Int, _ turnCount: Int) -> Bool {
-    if strikeCount == 3 {
+func isGameOver(_ strike: Int, _ inning: Int) -> Bool {
+    if strike == 3 {
         print("사용자 승리!")
         return true
     }
     
-    if turnCount == 0 {
+    if inning == 0 {
         print("컴퓨터 승리...!")
         return true
     }
     
-    print("남은 기회 : \(turnCount)")
+    print("남은 기회 : \(inning)")
     return false
 }
 
-func startNumberBaseball() {
-    var inning = 9
-    let computerNumberList = generateRandomNumberList()
+func verifyInputString(_ inputString: String?) -> [String]? {
+    guard let userNumberList = inputString?.components(separatedBy: .whitespaces) else { return  nil }
+    guard userNumberList.count == 3, Set(userNumberList).count == 3 else { return nil }
+    guard userNumberList.filter({
+        guard let integerNumber = Int($0), 1...9 ~= integerNumber else { return false }
+        
+        return true
+    }).count == 3 else { return nil }
+    
+    return userNumberList
+}
 
+func playNumberBaseball() {
+    let computerNumberList = generateRandomNumberList()
+    var inning = 9
+    
     while inning != 0 {
-        let randomNumberList = generateRandomNumberList()
-        let strikeAndBallCount = checkStrikeAndBall(computerNumberList, randomNumberList)
-
-        inning -= 1
-        print("""
-              임의의 수 : \(randomNumberList[0]) \(randomNumberList[1]) \(randomNumberList[2])
-              \(strikeAndBallCount.strike) 스트라이크, \(strikeAndBallCount.ball) 볼
-              """)
+        print("숫자 3개를 띄어쓰기로 구분하여 입력해주세요.\n중복 숫자는 허용하지 않습니다.\n입력", terminator: " : ")
+        let inputString = readLine()
+    
+        guard let userNumberList = verifyInputString(inputString) else {
+            print("입력이 잘못되었습니다.")
+            continue
+        }
         
+        let strikeAndBall = checkStrikeAndBall(computerNumberList, userNumberList)
+        inning -= 1
 
-        if isGameOver(strikeAndBallCount.strike, inning) {
+        if isGameOver(strikeAndBall.strike, inning) {
             break
         }
     }
 }
 
-startNumberBaseball()
+func start() {
+    while true {
+        print("1. 게임시작\n2. 게임종료\n원하는 기능을 선택해주세요", terminator: " : ")
+        let inputMenu = readLine()
+
+        switch inputMenu {
+        case "1":
+            playNumberBaseball()
+        case "2":
+            return
+        default:
+            print("입력이 잘못되었습니다.")
+        }
+    }
```

Comment:
while 반복문을 통해 게임시작, 종료 기능을 잘 구현해주셨네요 👍

저는 개인적으로 while의 반복 조건을 true로 고정해두기 보다는 Bool 타입 변수를 통해서 제어하는 것을 선호하는 편입니다.
한 번 참고삼아 읽어주세요 ☺️
```suggestion
    var shouldGameEnded = false
    
    while shouldGameEnded == false {
        print("1. 게임시작\n2. 게임종료\n원하는 기능을 선택해주세요", terminator: " : ")
        let inputMenu = readLine()

        switch inputMenu {
        case "1":
            playNumberBaseball()
        case "2":
            shouldGameEnded = true
        default:
            print("입력이 잘못되었습니다.")
        }
    }
```

Reply:
switch-case의 활용 👍👍👍

Reply:
좋은 아이디어 같습니다. 저희는 의논 끝에 변수를 추가하지 않는 방향으로 가기로 했습니다. 좋은 아이디어 감사합니다  😊